src/services/tracing.ts

```diff
@@ -171,6 +173,94 @@ export const asyncRunInSpan = <F extends () => ReturnType<F>>(
   return ret
 }
 
+const resolveFunctionSpanOptions = ({
+  namespace,
+  functionName,
+  functionArgs,
+  spanAttributes,
+}: {
+  namespace: string
+  functionName: string
+  functionArgs: Array<unknown>
+  spanAttributes?: SpanAttributes
+}): SpanOptions => {
+  const attributes = {
+    [SemanticAttributes.CODE_FUNCTION]: functionName,
+    [SemanticAttributes.CODE_NAMESPACE]: namespace,
+    ...spanAttributes,
+  }
+  if (functionArgs && functionArgs.length > 0) {
+    const params =
+      typeof functionArgs[0] === "object" ? functionArgs[0] : { "0": functionArgs[0] }
+    for (const key in params) {
+      attributes[`${SemanticAttributes.CODE_FUNCTION}.params.${key}`] = params[key]
+    }
+  }
+  return { attributes }
+}
+
+export const wrapToRunInSpan = <A extends Array<unknown>, R>({
+  fn,
+  namespace,
+  spanAttributes = {},
+}: {
+  fn: (...args: A) => R
+  namespace: string
+  spanAttributes?: SpanAttributes
+}) => {
+  return (...args: A): R => {
+    const functionName = fn.name
+    const spanName = `${namespace}.${functionName}`
+    const spanOptions = resolveFunctionSpanOptions({
+      namespace,
+      functionName,
+      functionArgs: args,
+      spanAttributes,
+    })
+    const ret = tracer.startActiveSpan(spanName, spanOptions, (span) => {
+      const ret = fn(...args)
+      if (ret instanceof Error) {
+        span.recordException(ret)
+      }
+      span.end()
+      return ret
+    })
+    return ret
+  }
+}
+
+type PromiseReturnType<T> = T extends Promise<infer Return> ? Return : T
+
+export const wrapAsyncToRunInSpan = <A extends Array<unknown>, R>({
+  fn,
+  namespace,
+  spanAttributes = {},
```

Comment:
this is never set in the auto instrumentation?

Reply:
I removed the param https://github.com/GaloyMoney/galoy/pull/837/commits/6c2545bd9bf949c6db30ea9355816b37cd355b8a